packages/core/src/internal/io.js

```diff
@@ -7,16 +7,19 @@ const TEST_HINT =
 
 const makeEffect = (type, payload) => ({ [IO]: true, type, payload })
 
-const isForkEffect = eff => eff && eff[IO] && eff.type === 'FORK'
-
 export const detach = eff => {
   if (process.env.NODE_ENV === 'development') {
+    const isForkEffect = eff => is.effect(eff) && eff.type === effectTypes.FORK
```

Comment:
why the change? why to move it to the function?

Reply:
`isForkEffect` is used only in development, so I move it inside the if-block to get better code elimination. 

However it turns out that even if `isForkEffect` is placed at top level, the code of `isForkEffect` would be removed when redux-saga builds. 😅

Reply:
yeah, in general function declarations and expressions are one of the easiest things to dead code eliminate, it's better to hoist this to the top level scope to avoid recreating this function inside each detach call